sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/DropFields.java

```diff
@@ -85,9 +85,8 @@ private Inner(FieldAccessDescriptor fieldsToDrop) {
 
     FieldAccessDescriptor complement(Schema inputSchema, FieldAccessDescriptor input) {
       // Create a FieldAccessDescriptor that select all fields _not_ selected in the input
-      // descriptor. Maintain
-      // the original order of the schema.
-      Set<String> fieldNamesToSelect = Sets.newHashSet();
+      // descriptor. Maintain the original order of the schema.
+      List<String> fieldNamesToSelect = newArrayList();
```

Comment:
Is this fixing a bug where field order isn't maintained? If so we should break it out into a separate PR

Reply:
And add a test

Reply:
You're definitely right, I've done it here:  https://github.com/apache/beam/pull/12928